taskmgr/queueservice_test.go

```diff
@@ -15,49 +19,736 @@ import (
+	service := queueService{
+		Log:              logger.WithField("component", "Queue Service"),
+		ExpirationOffset: 300,
+		Expires:          tcclient.Time(time.Now().Add(time.Minute * 10)),
+		queues: []taskQueue{
+			{
+				SignedDeleteUrl: fmt.Sprintf("%s/delete/{{messageId}}/{{popReceipt}}", s.URL),
+				SignedPollUrl:   fmt.Sprintf("%s/tasks/1234?messages=true", s.URL),
+			},
+			{
+				SignedDeleteUrl: fmt.Sprintf("%s/delete/{{messageId}}/{{popReceipt}}", s.URL),
+				SignedPollUrl:   fmt.Sprintf("%s/tasks/456?messages=true", s.URL),
+			},
+		},
+	}
+
+	tasks, err := service.retrieveTasksFromQueue(3)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, 3, len(tasks))
+	taskIds := []string{"abc", "def", "ghi"}
+	for i, v := range taskIds {
+		assert.Equal(t, v, tasks[i].TaskId)
+	}
+}
+
+func TestRetrieveTasksFromQueueDoesNotQueryLowPriority(t *testing.T) {
+	messages := []string{
+		// MessageText is {"taskId": "abc", "runId": 1}
+		`<?xml version="1.0" encoding="utf-8"?>
+	<QueueMessagesList>
+	  <QueueMessage>
+		<MessageId>5974b586-0df3-4e2d-ad0c-18e3892bfca3</MessageId>
+		<InsertionTime>Fri, 09 Oct 2009 21:04:30 GMT</InsertionTime>
+		<ExpirationTime>Fri, 16 Oct 2009 21:04:30 GMT</ExpirationTime>
+		<PopReceipt>YzQ4Yzg1MDItYTc0Ny00OWNjLTkxYTUtZGM0MDFiZDAwYzEw</PopReceipt>
+		<TimeNextVisible>Fri, 09 Oct 2009 23:29:20 GMT</TimeNextVisible>
+		<DequeueCount>1</DequeueCount>
+		<MessageText>eyJ0YXNrSWQiOiAiYWJjIiwgInJ1bklkIjogMX0=</MessageText>
+	  </QueueMessage>
+	</QueueMessagesList>`,
+		// MessageText[0] {"taskId": "def", "runId": 0}
+		// MessageText[1] {"taskId": "ghi", "runId": 2}
+		`<?xml version="1.0" encoding="utf-8"?>
+	<QueueMessagesList>
+	  <QueueMessage>
+		<MessageId>5974b586-0df3-4e2d-ad0c-18e3892bfca3</MessageId>
+		<InsertionTime>Fri, 09 Oct 2009 21:04:30 GMT</InsertionTime>
+		<ExpirationTime>Fri, 16 Oct 2009 21:04:30 GMT</ExpirationTime>
+		<PopReceipt>YzQ4Yzg1MDItYTc0Ny00OWNjLTkxYTUtZGM0MDFiZDAwYzEw</PopReceipt>
+		<TimeNextVisible>Fri, 09 Oct 2009 23:29:20 GMT</TimeNextVisible>
+		<DequeueCount>1</DequeueCount>
+		<MessageText>eyJ0YXNrSWQiOiAiZGVmIiwgInJ1bklkIjogMH0NCg==</MessageText>
+	  </QueueMessage>
+	  <QueueMessage>
+		<MessageId>5974b586-0df3-4e2d-ad0c-18e3892bfca3</MessageId>
+		<InsertionTime>Fri, 09 Oct 2009 21:04:30 GMT</InsertionTime>
+		<ExpirationTime>Fri, 16 Oct 2009 21:04:30 GMT</ExpirationTime>
+		<PopReceipt>YzQ4Yzg1MDItYTc0Ny00OWNjLTkxYTUtZGM0MDFiZDAwYzEw</PopReceipt>
+		<TimeNextVisible>Fri, 09 Oct 2009 23:29:20 GMT</TimeNextVisible>
+		<DequeueCount>1</DequeueCount>
+		<MessageText>eyJ0YXNrSWQiOiAiZ2hpIiwgInJ1bklkIjogMH0NCg==</MessageText>
+	  </QueueMessage>
+	</QueueMessagesList>`,
+	}
+
+	var handler = func(w http.ResponseWriter, r *http.Request) {
+		var message string
+		if r.URL.Path == "/tasks/1234" {
+			message = messages[1]
+		} else {
+			message = messages[0]
+		}
+		w.Header().Set("Content-Type", "application/xml")
+		w.Write([]byte(message))
+	}
```

Comment:
I think that there are a couple of handlers just returning a single xml string, and a couple that are verifying the delete endpoint was called (I guess we would need to make a package variable that stores that state rather than a variable in the test method then), and a few handlers that return some invalid things or a bad status code.